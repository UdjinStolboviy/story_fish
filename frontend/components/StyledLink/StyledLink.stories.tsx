import { expect } from "@storybook/jest";
import { screen } from "@storybook/testing-library";
import { ComponentStoryObj, ComponentMeta } from "@storybook/react";

import { StyledLink } from "./StyledLink";

export default {
  title: "Content/StyledLink",
  component: StyledLink,
} as ComponentMeta<typeof StyledLink>;

export const BasicStyledLink: ComponentStoryObj<typeof StyledLink> = {
  play: async () => {
    await expect(screen.getByRole("link")).toBeInTheDocument();
  },
  args: {
    children: "Hands-On React. Build advanced React JS Frontend with expert",
    href: "/hands-on-react-js",
  },
};
