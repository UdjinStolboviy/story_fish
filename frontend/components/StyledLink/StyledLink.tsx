import styled from "@emotion/styled";

export type Props = {
  underline?: boolean;
};

export const StyledLink = styled.a<Props>`
  all: unset;
  cursor: pointer;
  color: ${({ theme }) => theme.font.regular};
  text-decoration: ${({ underline }) => (underline ? "underline" : "none")};
  &:hover {
    opacity: 0.7;
  }
`;
