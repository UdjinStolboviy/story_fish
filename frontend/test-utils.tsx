import { FC, ReactElement } from "react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "@emotion/react";

import { rootReducer, RootState } from "./store";
import { Themes } from "./styles/themes";
import { Layout } from "./components/Layout";

type Options = { preloadedState?: RootState } & RenderOptions;

const customRender = (
  ui: ReactElement,
  { preloadedState, ...options }: Options = {}
) => {
  const store = configureStore({ reducer: rootReducer, preloadedState });

  const Wrapper: FC = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
    </Provider>
  );
  return render(ui, { wrapper: Wrapper, ...options });
};

const pageRender = (
  ui: ReactElement,
  { preloadedState, ...options }: Options = {}
) => {
  const store = configureStore({ reducer: rootReducer, preloadedState });

  const Wrapper: FC = ({ children }) => (
    <Provider store={store}>
      <Layout>{children}</Layout>
    </Provider>
  );
  return render(ui, { wrapper: Wrapper, ...options });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, pageRender };
