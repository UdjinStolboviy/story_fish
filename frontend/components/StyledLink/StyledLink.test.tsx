import { render } from "@/test-utils";

import { StyledLink } from "./StyledLink";

describe("StyledLink test cases", () => {
  it("StyledLink render check", () => {
    const { asFragment } = render(
      <StyledLink href="/hands-on-react-js">
        Hands-On React. Build advanced React JS Frontend with expert
      </StyledLink>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
