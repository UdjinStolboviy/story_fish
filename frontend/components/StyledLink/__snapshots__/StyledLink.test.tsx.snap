// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledLink test cases StyledLink render check 1`] = `
<DocumentFragment>
  <a
    class="css-1sdexyg"
    href="/hands-on-react-js"
  >
    Hands-On React. Build advanced React JS Frontend with expert
  </a>
</DocumentFragment>
`;
